tensorboard/webapp/plugins/npmi/store/npmi_reducers_test.ts

```diff
@@ -0,0 +1,220 @@
+    const state = createNpmiState();
+    const t0 = Date.now();
+    const nextState = reducers(
+      state,
+      actions.npmiAnnotationsLoaded({
+        annotations: {
+          run_1: ['annotation_1', 'annotation_2'],
+        },
+      })
+    );
+    expect(nextState.annotationsLoaded.state).toBe(DataLoadState.LOADED);
+    expect(
+      nextState.annotationsLoaded.lastLoadedTimeInMs
+    ).toBeGreaterThanOrEqual(t0);
+    expect(nextState.annotationsData).toEqual({
+      run_1: ['annotation_1', 'annotation_2'],
+    });
+  });
+
+  it('Overrides existing annotations on successful annotations loading', () => {
+    const state = createNpmiState({
+      annotationsData: {
+        run_1: ['annotation_1', 'annotation_2'],
+      },
+      annotationsLoaded: {
+        state: DataLoadState.LOADED,
+        lastLoadedTimeInMs: 0,
+      },
+    });
+    const t0 = Date.now();
+    const nextState = reducers(
+      state,
+      actions.npmiAnnotationsLoaded({
+        annotations: {
+          run_1: ['annotation_new_1', 'annotation_new_2'],
+        },
+      })
+    );
+    expect(nextState.annotationsLoaded.state).toBe(DataLoadState.LOADED);
+    expect(
+      nextState.annotationsLoaded.lastLoadedTimeInMs
+    ).toBeGreaterThanOrEqual(t0);
+    expect(nextState.annotationsData).toEqual({
+      run_1: ['annotation_new_1', 'annotation_new_2'],
+    });
+  });
+});
+
+describe('Metrics and Values loading', () => {
+  it('sets metricsAndValuesLoaded to loading on requesting', () => {
+    const state = createNpmiState();
+    const nextState = reducers(state, actions.npmiMetricsAndValuesRequested());
+    expect(nextState.metricsAndValuesLoaded.state).toBe(DataLoadState.LOADING);
+    expect(nextState.metricsAndValuesLoaded.lastLoadedTimeInMs).toBeNull();
+  });
+
+  it('set metricsLoaded to failed on request failure', () => {
+    const state = createNpmiState({
+      metricsAndValuesLoaded: {
+        state: DataLoadState.LOADING,
+        lastLoadedTimeInMs: null,
+      },
+    });
+    const nextState = reducers(
+      state,
+      actions.npmiMetricsAndValuesRequestFailed()
+    );
+    expect(nextState.metricsAndValuesLoaded.state).toBe(DataLoadState.FAILED);
+    expect(nextState.metricsAndValuesLoaded.lastLoadedTimeInMs).toBeNull();
+  });
+
+  it('sets metricsAncValuesLoaded, metrics & values on successful load', () => {
+    const state = createNpmiState();
+    const t0 = Date.now();
+    const nextState = reducers(
+      state,
+      actions.npmiMetricsAndValuesLoaded({
+        metrics: {
+          run_1: [
+            'count',
+            'count@test1',
+            'count@test2',
+            'nPMI@test1',
+            'nPMI@test2',
+          ],
+        },
+        values: {
+          run_1: [
+            [3510517, 16719, 513767, 0.16871, -0.37206],
+            [1396813, 1896, 638967, 0.687616, 0.68116],
+          ],
+        },
+      })
```

Comment:
Make this test interesting by varying the response. Few ideas:

1. `metrics['run_1'].length !== values['run_1'].length`
2. Change the order of the runs around. e.g., 
```
'count@test1',
'count',
'nPMI@test1',
'count@test2',
'nPMI@test2
```